File: app/schemas/tipo_ocorrencia.py
"""Pydantic schemas for TipoOcorrencia entity."""

from datetime import datetime

from pydantic import BaseModel, Field


class TipoOcorrenciaBase(BaseModel):
    nome: str = Field(..., max_length=120)
    descricao: str | None = None


class TipoOcorrenciaCreate(TipoOcorrenciaBase):
    pass


class TipoOcorrenciaUpdate(BaseModel):
    nome: str | None = Field(None, max_length=120)
    descricao: str | None = None


class TipoOcorrenciaResponse(TipoOcorrenciaBase):
    id: int
    ativo: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/tipo-ocorrencia.ts
/** TypeScript interfaces for TipoOcorrencia entity matching Pydantic schemas. */

export interface TipoOcorrenciaResponse {
  id: number;
  nome: string;
  descricao: string | null;
  ativo: boolean;
  created_at: string;
  updated_at: string;
}

export interface TipoOcorrenciaCreate {
  nome: string;
  descricao?: string | null;
}

export interface TipoOcorrenciaUpdate {
  nome?: string | null;
  descricao?: string | null;
}
